src/main/java/mate/academy/dao/impl/BookDaoImpl.java

```diff
@@ -0,0 +1,116 @@
+            if (resultSet.next()) {
+                Long id = resultSet.getLong(1);
+                book.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to create a book = [" + book + "]", e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String query = "SELECT * FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                Book book = getBookFromResultSet(resultSet);
+                return Optional.of(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to find a book with id = [" + id + "]", e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        String query = "SELECT * FROM books";
+        List<Book> books = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Book book = getBookFromResultSet(resultSet);
+                books.add(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to fetch all books", e);
+        }
+        return books;
+    }
+
+    @Override
+    public Book update(Book book) {
+        String query = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.setLong(3, book.getId());
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows < 1) {
+                throw new DataProcessingException("Expected to affect at least one row, "
+                        + "but got affected = [" + affectedRows + "]");
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to update a book = [" + book + "]", e);
+        }
+        return book;
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        String query = "DELETE FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
```

Comment:
Instead of returning `true` all the time, let's return a boolean value depending on the `preparedStatement.executeUpdate()` result. This makes the method's behavior more accurate to whether the deletion actually happened.